internal/metric/wrapper.go

```diff
@@ -0,0 +1,208 @@
+package metric
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	monitoring "cloud.google.com/go/monitoring/apiv3"
+	"github.com/pkg/errors"
+	"google.golang.org/api/iterator"
+	monitoringpb "google.golang.org/genproto/googleapis/monitoring/v3"
+	duration "google.golang.org/protobuf/types/known/durationpb"
+	timestamp "google.golang.org/protobuf/types/known/timestamppb"
+)
+
+// Client represents a wrapper around the Cloud Monitoring package.
+type Client interface {
+	Latency(ctx context.Context, filter Filter, startTime time.Time) (float64, error)
+}
+
+// API is a wrapper for the Cloud Monitoring package.
+type API struct {
+	Client  *monitoring.MetricClient
+	Project string
+}
+
+// Filter is the filter for the query.
+type Filter struct {
+	service  string
+	revision string
+	query    string
+}
+
+// Metric types.
+const (
```

Comment:
might be wise to call this package `stackdriver`, and have `metrics` package just be the generic interfaces (so you can replace stackdriver with mocks etc).